### PR TITLE
Fix flare e2e test

### PR DESCRIPTION
**Pull request by Kaderinho**
### What does this PR do?

Fix failing e2e tests in the CI

### Motivation

Failing tests are bad

### Possible Drawbacks / Trade-offs

Sleep is not the best solution. It will need a change when the framework will provide tool to verify fakeintake readiness

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/fakeintake/server/server.go [hunk 1]
```diff
@@ -246,14 +246,6 @@ func (fi *Server) handleDatadogRequest(w http.ResponseWriter, req *http.Request)
...
 	}
 
-	// Datadog Agent sends a HEAD request to avoid redirect issue before sending the actual flare
-	if req.Method == http.MethodHead && req.URL.Path == "/support/flare" {
-		writeHTTPResponse(w, httpResponse{
-			statusCode: http.StatusOK,
-		})
-		return
-	}
-
```
Comment (pgimalac) [inline test/fakeintake/server/server.go:256]: **❓ question:** ‏Why remove this ?

File: test/new-e2e/agent-subcommands/flare/flare_test.go [hunk 2]
```diff
@@ -27,6 +28,9 @@ func TestFlareSuite(t *testing.T) {
 }
 
 func requestAgentFlareAndFetchFromFakeIntake(v *commandFlareSuite, flareArgs ...client.AgentArgsOption) flare.Flare {
+	// Wait for the fakeintake to be ready to avoid 503 when sending the flare
+	// Definitely not the best way to do it, but the easiest at the moment
+	time.Sleep(60 * time.Second)
```
Comment (pgimalac) [inline test/new-e2e/agent-subcommands/flare/flare_test.go:33]: **❓ question:** ‏Any reason not to use an eventually (eg. eventually some basic metric is received by the fakeintake) ?

**Review by pgimalac — APPROVED** [review]

File: test/fakeintake/server/server.go (hunk 1 above)
Comment (Kaderinho) [reply test/fakeintake/server/server.go:256]: It's duplicated above, probably after a wrong merge resolution

File: test/new-e2e/agent-subcommands/flare/flare_test.go (hunk 2 above)
Comment (Kaderinho) [reply test/new-e2e/agent-subcommands/flare/flare_test.go:33]: Was easier this way, but yes the solution you described is way better. I'll change that when the framework will have a function to easily check that the fakeintake is ready

Comment (pducolin) [reply test/new-e2e/agent-subcommands/flare/flare_test.go:33]: **💬 suggestion**
You can test that the `fakeintake` is ready by checking you get a `StatusOK` response on `v.Env().Fakeintake.Client.GetServerHealth()` --> https://github.com/DataDog/datadog-agent/blob/main/test/fakeintake/client/client.go#L148-L158

You could do something like 

```golang
assert.EventuallyWithT(t, func(c *assert.CollectT) {
    assert.NoErr(c, v.Env().Fakeintake.Client.GetServerHealth())
}) 
```

File: test/new-e2e/agent-subcommands/flare/flare_test.go [hunk 3]
```diff
@@ -27,10 +29,15 @@ func TestFlareSuite(t *testing.T) {
 }
 
 func requestAgentFlareAndFetchFromFakeIntake(v *commandFlareSuite, flareArgs ...client.AgentArgsOption) flare.Flare {
+	// Wait for the fakeintake to be ready to avoid 503 when sending the flare
+	v.EventuallyWithT(func(c *assert.CollectT) {
```
Comment (ogaca-dd) [inline test/new-e2e/agent-subcommands/flare/flare_test.go:33]: @pducolin : What do you think about moving this code to `test-infra-definition` to make sure fakeintake is ready when the user have access to `v.Env().Fakeintake`?

**Review by ogaca-dd — APPROVED** [review]

**Pull request merged**